esmvalcore/preprocessor/_regrid.py

```diff
@@ -306,6 +307,48 @@ def _attempt_irregular_regridding(cube, scheme):
     return False
 
 
+def extract_location(cube, location, scheme):
+    """Extract a point using a location name, with interpolation.
+
+    Extracts a single location point from a cube, according
+    to the interpolation scheme `scheme`.
+
+    The function just retrieves the coordinates of the location and then calls
+    the `extract_point` preprocessor.
+
+    It can be use to locate cities and villages, but also mountains or other
+    geographical locations.
+
+    Parameters
+    ----------
+    cube : cube
+        The source cube to extract a point from.
+
+    location : str
+        The reference location.
+
+    scheme : str
+        The interpolation scheme. 'linear' or 'nearest'. No default.
+
+    Returns
+    -------
+    Returns a cube with the extracted point, and with adjusted
+    latitude and longitude coordinates.
+
+    Raises
+    ------
+    ValueError:
+        If given location can not be found by the geolocator.
+    """
+    geolocator = Nominatim(user_agent='esmvalcore')
+    location = geolocator.geocode(location)
+    if not location:
+        raise ValueError(f'Requested location {location} can not be found.')
```

Comment:
You need to assign another variable name for `location`, if the location is not find I get the following error:

```
ValueError: Requested location None can not be found.
```

And maybe check for `if ... is not None` instead of `if not ...`?

Reply:
ahem, will do, a check if `None` is prob enough here

Reply:
Ah, sorry, I wasn't specific enough:

If the location cannot be found, `location` is set to `None` in l.357. This leads to the error message `ValueError: Requested location None can not be found.`, regardless of the location specified. I would like to have a message like `ValueError: Requested location Gondor can not be found.`